src/components/search/SearchResults.tsx

```diff
@@ -0,0 +1,190 @@
+  // cannot use onBlur on Search because it is fired first and the click in the list is canceled
+  const ref = useOutsideClick<HTMLUListElement>(() => {
+    onOutsideClick?.(false);
+  });
+  const theme = useTheme();
+  const router = useRouter();
+  const { hooks } = useContext(QueryClientContext);
+  const { data: collections } = hooks.useSearchPublishedItems({
+    query,
+    categories,
+    isPublishedRoot,
+    enabled: Boolean(query),
+    cropLength: 10,
+    attributesToCrop: ['content', 'description', 'name'],
+    highlightPostTag: '</span>',
+    highlightPreTag: `<span style="font-weight:bold;color:${theme.palette.primary.main}">`,
+  });
+
+  const onLoadMoreClick = () => {
+    // navigate to "/all-collections?cat=<categoryId>"
+    router.push({
+      pathname: ALL_COLLECTIONS_ROUTE,
+      query: { [UrlSearch.KeywordSearch]: query },
+    });
+  };
+
+  if (!collections) {
+    return null;
+  }
+  // eslint-disable-next-line prefer-destructuring
+  const queryResults = collections.results.first();
+  const hits = queryResults?.hits;
+  // no result found
+  if (!hits || hits.isEmpty()) {
+    return (
+      <Container
+        sx={{
+          py: 1,
+          pl: 2,
+        }}
+      >
+        <Trans t={t} i18nKey={LIBRARY.SEARCH_NO_RESULTS} />
+      </Container>
+    );
+  }
+
+  return (
+    <Container>
+      <List id={SEARCH_RESULTS_LIST_ID} sx={{ width: '100%' }} ref={ref}>
+        {hits.take(MAX_RESULTS_TO_SHOW).map((result) => (
+          <>
+            <ListItemButton
+              component="a"
+              href={buildCollectionRoute(result.id)}
+              target="_blank"
+            >
+              <ListItemText
+              // primary={result._formatted.name}
+              >
+                <Stack direction="row" alignItems="center">
+                  <Stack>
+                    <SearchThumbnail name={result.name} itemId={result.id} />
+                  </Stack>
+                  <Stack>
+                    <Stack direction="column">
+                      <Interweave
+                        style={{ paddingRight: 5 }}
+                        //  eslint-disable-next-line no-underscore-dangle
+                        content={result._formatted.name}
+                      />
+                    </Stack>
+                    <Stack>
+                      <Interweave
+                        style={{ color: '#999' }}
+                        content={
+                          // eslint-disable-next-line no-underscore-dangle
+                          result._formatted.description ||
+                          // eslint-disable-next-line no-underscore-dangle
+                          result._formatted.content
+                        }
+                      />
+                    </Stack>
+                  </Stack>
+                </Stack>
+              </ListItemText>
+            </ListItemButton>
+            <Divider />
```

Comment:
To avoid the "last divider" problem and make it more concise we could use something like [this `intersperse()` utility function](https://github.com/mui/material-ui/issues/18158#issuecomment-748602867)